core/storage/base_model/gae_models.py

```diff
@@ -542,17 +540,34 @@ def get_multi_versions(cls, entity_id, version_numbers):
         Returns:
             list(VersionedModel). Model instances representing the given
                 versions.
+
+        Raises:
+            ValueError. Given entity_id is invalid.
+            ValueError. Requested version number cannot be higher than the
+                current version number.
+            ValueError. Version number is invalid.
         """
         instances = []
-        try:
-            cls.get(entity_id)
-        except Exception as  e:
-            logging.error(e)
+
+        entity = cls.get(entity_id, strict=False)
+        if not entity:
+            raise ValueError('Given entity_id %s is invalid.' % (entity_id))
+        current_version = entity.version
+        if version_numbers[-1] > current_version:
+            raise ValueError(
+                'Requested version number cannot be higher than the current '
+                'version number.')
+
         # pylint: disable=protected-access
         for version in version_numbers:
             snapshot_id = cls._get_snapshot_id(entity_id, version)
-            instances.append(cls(id=entity_id)._reconstitute_from_snapshot_id(
-                snapshot_id))
+            try:
+                entity = cls(id=entity_id)._reconstitute_from_snapshot_id(
```

Comment:
I just realized that this is basically a get() in a for loop. I think you'll need a "multi" version of reconstitute as well (or maybe you can extract the get part from it, do the get outside reconstitute(), and pass the data in).

Reply:
Done.